src/Components/Blazor/testassets/Wasm.Authentication.Client/Shared/LoginDisplay.razor

```diff
@@ -1,21 +1,15 @@
 @using Microsoft.AspNetCore.Components.Authorization
-@inject NavigationManager Navigation
-@inject SignOutSessionStateManager SignOutManager
-<AuthorizeView>
+<AuthorizeView Context="authenticationState">
     <Authorized>
-        <a href="authentication/profile">Hello, @context.User.Identity.Name!</a>
-        <button class="nav-link btn btn-link" @onclick="BeginLogout">Log out</button>
+        <a href="authentication/profile">Hello, @authenticationState.User.Identity.Name!</a>
+        <SignOut>
+            <OnSignOut Context="BeginSignOut">
+                <button class="nav-link btn btn-link" @onclick="BeginSignOut">Log out</button>
+            </OnSignOut>
+        </SignOut>
```

Comment:
This is exceptionally unusual, and I don't really feel comfortable with it. It's much more elaborate than people would expect just to have a "sign out" button.

Let's discuss later what's needed to make sign out work and what other options might exist.

Reply:
Updated as per our discussion.